packages/paste-core/components/popover/src/PopoverButton.tsx

```diff
@@ -1,26 +1,40 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import type {ButtonProps} from '@twilio-paste/button';
-import type {BoxProps} from '@twilio-paste/box';
 import {Button} from '@twilio-paste/button';
+import {Badge} from '@twilio-paste/badge';
 import {NonModalDialogDisclosurePrimitive} from '@twilio-paste/non-modal-dialog-primitive';
+import type {PopoverButtonProps, ButtonBadgeProps} from './types';
 import {PopoverContext} from './PopoverContext';
 
-export interface PopoverButtonProps extends ButtonProps, Pick<BoxProps, 'element'> {
-  id?: string;
-  children: React.ReactNode;
-  toggle?: () => void;
-}
+// eslint-disable-next-line react/display-name
+const ButtonBadge = React.forwardRef<HTMLButtonElement, ButtonBadgeProps>(({children, ...props}, ref) => (
+  <Badge {...props} as="button" ref={ref}>
+    {children}
+  </Badge>
+));
 
 const PopoverButton = React.forwardRef<HTMLButtonElement, PopoverButtonProps>(
-  ({children, element = 'POPOVER_BUTTON', ...popoverButtonProps}, ref) => {
+  ({as = 'button', children, element = 'POPOVER_BUTTON', ...popoverButtonProps}, ref) => {
     const popover = React.useContext(PopoverContext);
+    let ButtonComponent;
+
+    if (as === 'button') {
+      ButtonComponent = Button;
+    } else if (as === 'badge') {
+      ButtonComponent = ButtonBadge;
+    }
+
+    let computedElement = element;
+    if (as === 'badge' && element === 'POPOVER_BUTTON') {
+      computedElement = 'POPOVER_BADGE';
```

Comment:
I thought it wouldn't.... 'POPOVER_BUTTON' is the default value, so if you pass something that isn't the default value and it is a badge it will get passed.

Reply:
I think a test that covers the logic would prove it either way. 😉

Reply:
Fair point, added tests